cupy/manipulation/kind.py

```diff
@@ -24,4 +25,55 @@ def asfortranarray(a, dtype=None):
 # TODO(okuta): Implement asarray_chkfinite
 
 
-# TODO(okuta): Implement require
+def require(a, dtype=None, requirements=None):
+    """Return an array which satisfies the requirements.
+
+    Args:
+        a (~cupy.ndarray): The input array.
+        dtype (str or dtype object, optional): The required data-type.
+            If None preserve the current dtype.
+        requirements (str or list of str): The requirements can be any
+            of the following
+
+            * 'F_CONTIGUOUS' ('F', 'FORTRAN') - ensure a Fortran-contiguous \
+                array. \
+
+            * 'C_CONTIGUOUS' ('C', 'CONTIGUOUS') - ensure a C-contiguous array.
+
+            * 'OWNDATA' ('O')      - ensure an array that owns its own data.
+
+    Returns:
+        ~cupy.ndarray: The input array ``a`` with specified requirements and
+            type if provided.
+
+    .. seealso:: :func:`numpy.require`
+
+    """
+
+    possible_flags = {'C': 'C', 'C_CONTIGUOUS': 'C', 'CONTIGUOUS': 'C',
+                      'F': 'F', 'F_CONTIGUOUS': 'F', 'FORTRAN': 'F',
+                      'O': 'OWNDATA', 'OWNDATA': 'OWNDATA'}
+
+    if not requirements:
+        return cupy.asanyarray(a, dtype=dtype)
+    else:
+        requirements = {possible_flags[x.upper()] for x in requirements}
```

Comment:
Lets raise a value error with a descriptive message if the requirement is not in the flags keys.